### PR TITLE
[Mapping] adding readme

### PR DESCRIPTION
**Pull request by philbucher**
@pooyan-dadvand while preparing the example for the tutorial of the CoSimIO I thought it would make more sense to add some proper readme to Kratos itself and refer to it

### CODE REVIEW

File: applications/MappingApplication/README.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+The **Map** function is used to map values from the **Origin** to the **Destination**. For this the _Variables_ have to be specified. See the following example for mapping scalar quantities.
+
+**Note**: In order to demonstrate that the _Mapper_ is not tied to any particular physics, arbitrary _Variables_ are chosen in the following examples.
+
+~~~py
+# mapping scalar quantities
+# this maps the nodal quantities of TEMPERATURE on the origin-ModelPart
+# to the nodal quantities of PRESSURE on the destination-ModelPart
+
+mapper.Map(KM.TEMPERATURE, KM.PRESSURE)
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md]:
```suggestion
mapper.Map(KM.TEMPERATURE, KM.AMBIENT_TEMPERATURE)
```

Just to have more phisical meaning

File: applications/MappingApplication/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+~~~
+
+The **Map** function is overloaded, this means that mapping vector quantities works in the same way as mapping scalar quantites.
+
+~~~py
+# mapping vector quantities
+# this maps the nodal quantities of VELOCITY on the origin-ModelPart
+# to the nodal quantities of DISPLACEMENT on the destination-ModelPart.
+
+mapper.Map(KM.VELOCITY, KM.DISPLACEMENT)
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md]:
```suggestion
mapper.Map(KM.MESH_VELOCITY, KM.VELOCITY)
```

File: applications/MappingApplication/README.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+Internally it constructs the mapping matrix, hence it offers the usage of the transposed mapping matrix. When using this, for very inhomogenous interface discretizations it can come to oscillations in the mapped quantities.
+
+#### Nearest Element
+The _NearestElementMapper_ projects nodes to the elements( or conditions) on other side of the inteface. Mapping is then done by interpolating the values of the nodes of the elements by using the shape functions at the projected position.
+
+This mapper is best suited for problems where the _NearestNeighborMapper_ cannot be used, i.e. for cases where the discretization on the interfaces is different. Note that it is less robust than the _NearestNeighborMapper_ due to the projections it performs. In case a projection fails it uses an approximation that is similar to the approach of the _NearestNeighborMapper_.
+
+Internally it constructs the mapping matrix, hence it offers the usage of the transposed mapping matrix. When using this, for very inhomogenous interface discretizations it can come to oscillations in the mapped quantities.
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md:187]: Is not very clear if the oscillation happens when using transpose or not.

File: applications/MappingApplication/README.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
+## Mapping Application
+The Mapping Application contains the core developments in mapping data between non matching grids. It works both in shared and distributed (**MPI**) memory environments as well as in 1D, 2D and 3D domains.
+
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md:3]: I would make the feature list more visible:
* Serial/OpenMP/MPI
* 1D/2D/3D
* Matching/NonMatching
* Mappers (Nearest Neighbors, ...)
* Mapping operations: (ADD_VALUES, SWAP_SIGN, etc.)

I would also add a list of supported mesh topologies (shapes, linear/quadratic, ...)

File: applications/MappingApplication/README.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+~~~
+
+Mapping from **Destination** to **Origin** can be done using the **InverseMap** function which works in the same way as the **Map** function.
+
+~~~py
+# inverse mapping scalar quantities
+# this maps the nodal quantities of PRESSURE on the destination-ModelPart
+# to the nodal quantities of TEMPERATURE on the origin-ModelPart
+
+mapper.InverseMap(KM.TEMPERATURE, KM.PRESSURE)
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md]:
```suggestion
mapper.InverseMap(KM.TEMPERATURE, KM.AMBIENT_TEMPERATURE)
```

File: applications/MappingApplication/README.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+# inverse mapping vector quantities
+# this maps the nodal quantities of DISPLACEMENT on the destination-ModelPart
+# to the nodal quantities of VELOCITY on the origin-ModelPart
+
+mapper.InverseMap(KM.VELOCITY, KM.DISPLACEMENT)
```
Comment (pooyan-dadvand) [inline applications/MappingApplication/README.md]:
```suggestion
mapper.InverseMap(KM.EMBEDDED_VELOCITY, KM.VELOCITY)
```

**Review by pooyan-dadvand — COMMENTED** [review]
Very nice overview!! Only some minor comments

File: applications/MappingApplication/README.md (hunk 4 above)
Comment (philbucher) [reply applications/MappingApplication/README.md:3]: done

File: applications/MappingApplication/README.md (hunk 3 above)
Comment (philbucher) [reply applications/MappingApplication/README.md:187]: clarified

File: applications/MappingApplication/README.md (hunk 5 above)
Comment (philbucher) [reply applications/MappingApplication/README.md]: done

File: applications/MappingApplication/README.md (hunk 6 above)
Comment (philbucher) [reply applications/MappingApplication/README.md]: done

File: applications/MappingApplication/README.md (hunk 3 above)
Comment (philbucher) [reply applications/MappingApplication/README.md:187]: clarified

File: applications/MappingApplication/README.md (hunk 2 above)
Comment (philbucher) [reply applications/MappingApplication/README.md]: done

File: applications/MappingApplication/README.md (hunk 1 above)
Comment (philbucher) [reply applications/MappingApplication/README.md]: done

**Review by pooyan-dadvand — APPROVED** [review]
:+1:

**Comment by ddiezrod** [comment]
Is search radius not needed anymore?

**Comment by philbucher** [comment]
> Is search radius not needed anymore? @philbucher

it is for now
However I am currently redesigning the search since it takes too much time and memory and I didn't want to add things to the readme that will change soon

Once the new search is done I will add it too

(for the same reason I also didn't add the explanation how to map non-historical values yet)

**Comment by ddiezrod** [comment]
Ok! Thanks @philbucher , when you upload the changes please ping me if you recall.

**Pull request merged**